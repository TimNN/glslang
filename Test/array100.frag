#version 100

float gu[];              // ERROR
float g4[4];
float g5[5];

uniform int a;

float[4] foo(float[5] a)  // ERROR  // ERROR
{
    return float[](a[0], a[1], a[2], a[3]);  // ERROR
}

void bar(float[5]) {}

void main()
{
    {
        float gu[2];  // okay, new scope

        gu[2] = 4.0;  // ERROR, overflow
    }

    g4 = foo(g5);
    g5 = g4;  // ERROR
    gu = g4;  // ERROR

    foo(gu);  // ERROR
    bar(g5);

    if (float[4](1.0, 2.0, 3.0, 4.0) == g4)  // ERROR
        gu[0] = 2.0;

    float u[5];
    u[5] = 5.0; // ERROR
    foo(u);     // okay

    gl_FragData[1000] = vec4(1.0); // ERROR
    gl_FragData[-1] = vec4(1.0);   // ERROR
    gl_FragData[3] = vec4(1.0);
}
