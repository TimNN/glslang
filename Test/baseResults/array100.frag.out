ERROR: 0:3: 'float' : type requires declaration of default precision qualifier 
ERROR: 0:3: '' : array size required 
ERROR: 0:9: 'arrayed type' : not supported for this version or the enabled extensions 
ERROR: 0:9: 'arrayed type' : not supported for this version or the enabled extensions 
ERROR: 0:11: 'arrayed constructor' : not supported for this version or the enabled extensions 
ERROR: 0:21: '[' :  array index out of range '2'
ERROR: 0:25: 'assign' :  cannot convert from '4-element array of mediump float' to '5-element array of mediump float'
ERROR: 0:26: 'assign' :  cannot convert from '4-element array of mediump float' to 'unsized array of mediump float'
ERROR: 0:28: 'foo' : no matching overloaded function found 
ERROR: 0:31: 'arrayed constructor' : not supported for this version or the enabled extensions 
ERROR: 0:35: '[' :  array index out of range '5'
ERROR: 0:38: '[' :  array index out of range '1000'
ERROR: 0:39: '[' :  array index out of range '-1'
ERROR: 13 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:9  Function Definition: foo(f1[5]; (4-element array of mediump float)
0:9    Function Parameters: 
0:9      'a' (in 5-element array of mediump float)
0:11    Sequence
0:11      Branch: Return with expression
0:11        Construct float (4-element array of float)
0:11          direct index (in mediump float)
0:11            'a' (in 5-element array of mediump float)
0:11            Constant:
0:11              0 (const int)
0:11          direct index (in mediump float)
0:11            'a' (in 5-element array of mediump float)
0:11            Constant:
0:11              1 (const int)
0:11          direct index (in mediump float)
0:11            'a' (in 5-element array of mediump float)
0:11            Constant:
0:11              2 (const int)
0:11          direct index (in mediump float)
0:11            'a' (in 5-element array of mediump float)
0:11            Constant:
0:11              3 (const int)
0:14  Function Definition: bar(f1[5]; (void)
0:14    Function Parameters: 
0:14      '' (in 5-element array of mediump float)
0:16  Function Definition: main( (void)
0:16    Function Parameters: 
0:?     Sequence
0:?       Sequence
0:21        move second child to first child (mediump float)
0:21          direct index (mediump float)
0:21            'gu' (2-element array of mediump float)
0:21            Constant:
0:21              2 (const int)
0:21          Constant:
0:21            4.000000
0:24      move second child to first child (4-element array of mediump float)
0:24        'g4' (4-element array of mediump float)
0:24        Function Call: foo(f1[5]; (4-element array of mediump float)
0:24          'g5' (5-element array of mediump float)
0:25      'g5' (5-element array of mediump float)
0:26      'gu' (unsized array of mediump float)
0:28      Constant:
0:28        0.000000
0:29      Function Call: bar(f1[5]; (void)
0:29        'g5' (5-element array of mediump float)
0:31      Test condition and select (void)
0:31        Condition
0:31        Compare Equal (bool)
0:31          Constant:
0:31            1.000000
0:31            2.000000
0:31            3.000000
0:31            4.000000
0:31          'g4' (4-element array of mediump float)
0:31        true case
0:32        move second child to first child (mediump float)
0:32          direct index (mediump float)
0:32            'gu' (unsized array of mediump float)
0:32            Constant:
0:32              0 (const int)
0:32          Constant:
0:32            2.000000
0:35      move second child to first child (mediump float)
0:35        direct index (mediump float)
0:35          'u' (5-element array of mediump float)
0:35          Constant:
0:35            5 (const int)
0:35        Constant:
0:35          5.000000
0:36      Function Call: foo(f1[5]; (4-element array of mediump float)
0:36        'u' (5-element array of mediump float)
0:38      move second child to first child (4-component vector of float)
0:38        direct index (fragColor 4-component vector of float)
0:38          'gl_FragData' (fragColor 32-element array of 4-component vector of float)
0:38          Constant:
0:38            1000 (const int)
0:38        Constant:
0:38          1.000000
0:38          1.000000
0:38          1.000000
0:38          1.000000
0:39      move second child to first child (4-component vector of float)
0:39        direct index (fragColor 4-component vector of float)
0:39          'gl_FragData' (fragColor 32-element array of 4-component vector of float)
0:39          Constant:
0:39            -1 (const int)
0:39        Constant:
0:39          1.000000
0:39          1.000000
0:39          1.000000
0:39          1.000000
0:40      move second child to first child (4-component vector of float)
0:40        direct index (fragColor 4-component vector of float)
0:40          'gl_FragData' (fragColor 32-element array of 4-component vector of float)
0:40          Constant:
0:40            3 (const int)
0:40        Constant:
0:40          1.000000
0:40          1.000000
0:40          1.000000
0:40          1.000000
0:?   Linker Objects
0:?     'gu' (unsized array of mediump float)
0:?     'g4' (4-element array of mediump float)
0:?     'g5' (5-element array of mediump float)
0:?     'a' (uniform mediump int)

